lib/Rx/Observable/TimerObservable.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Rx\Observable;
+
+use Rx\Observable;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+class TimerObservable extends Observable
+{
+    /** @var int */
+    private $dueTime;
+
+    /** @var int */
+    private $period;
+
+    /** @var SchedulerInterface */
+    private $scheduler;
+
+    public function __construct($dueTime, $period = 0, SchedulerInterface $scheduler = null)
+    {
+
+        if (!is_int($dueTime) || !is_int($period)) {
+            throw new \InvalidArgumentException("'dueTime' and 'period' must be an integer");
+        }
+
+        if ($period < 0) {
+            throw new \InvalidArgumentException("'period' cannot be a negative number");
+        }
+
+        $this->dueTime   = $dueTime;
+        $this->period    = $period;
+        $this->scheduler = $scheduler;
+    }
+
+
+    public function subscribe(ObserverInterface $observer, SchedulerInterface $scheduler = null)
+    {
+        if ($this->scheduler !== null) {
+            $scheduler = $this->scheduler;
+        }
+        if ($scheduler === null) {
+            throw new \Exception("You must use a scheduler that support non-zero delay.");
+        }
+
+        $counter = 0;
+
+        if ($this->period > 0) {
+            $disposable = $scheduler->schedulePeriodic(
+                function () use (&$counter, $observer) {
+                    $observer->onNext($counter++);
+                },
+                $this->dueTime,
+                $this->period
+            );
+        } else {
+            $disposable = $scheduler->schedule(
+                function () use (&$counter, $observer) {
+                    $observer->onNext($counter);
+                    $observer->onCompleted();
```

Comment:
I think we do this in other places as well. We can revisit it later. Good question though.